packages/paste-core/components/tabs/src/Tab.tsx

```diff
@@ -0,0 +1,111 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import {Box, safelySpreadBoxProps} from '@twilio-paste/box';
+import {Text} from '@twilio-paste/text';
+import {TabPrimitive} from '@twilio-paste/tabs-primitive';
+import {TabsContext} from './TabsContext';
+
+type Orientation = 'horizontal' | 'vertical' | undefined;
+
+const getTabBoxStyles = (orientation: Orientation): {} => {
+  switch (orientation) {
+    case 'vertical':
+      return {
+        display: 'block',
+        right: '-2px',
+        paddingBottom: 'space40',
+        paddingTop: 'space40',
+        paddingLeft: 'space50',
+        paddingRight: 'space50',
+        borderRightWidth: 'borderWidth20',
+        borderRightStyle: 'solid',
+        borderRightColor: 'transparent',
+        _selected: {
+          borderRightColor: 'colorBorderPrimary',
+        },
+      };
+    case 'horizontal':
+    default:
+      return {
+        display: 'inline-block',
+        bottom: '-2px',
+        paddingBottom: 'space10',
+        paddingTop: 'space20',
+        paddingLeft: 'space50',
+        paddingRight: 'space50',
+        borderBottomWidth: 'borderWidth20',
+        borderBottomStyle: 'solid',
+        borderBottomColor: 'transparent',
+        _selected: {
+          borderBottomColor: 'colorBorderPrimary',
+        },
+      };
+  }
+};
+
+const getTabTextStyles = (props: TabProps): {} => {
```

Comment:
Should the return type here be `TextColor` or `TextColorOptions` from our style props?

Reply:
I just changed this to also include _hover, so I'm going to leave it ambiguous for now.